File: 0169-majority-element/0169-majority-element.rb
# @param {Integer[]} nums
# @return {Integer}
def majority_element(nums)
    res = nums[0]
    cnt = 1
    nums[1..-1].each do |num|
        if cnt == 0
            res = num
            cnt = 1
            next
        end

        if num == res
            cnt += 1
        else
            cnt -= 1
        end
    end
    
    res
end

File: 0169-majority-element/0169-majority-element.go
func majorityElement(nums []int) int {
    major := nums[0]
    cnt := 1
    for i := 1; i < len(nums); i++ {
        if cnt == 0 {
            major = nums[i]
            cnt = 1
            continue
        }

        if nums[i] == major {
            cnt++
        } else {
            cnt--
        }
    }

    return major
}